File: semester2/5_Fuzzy-Software-Technologies/aics-fuzzy-software-technologies/ticket-backend/src/main/java/aics/domain/fuzzy/dtos/FuzzyProfileDto.java
package aics.domain.fuzzy.dtos;

import aics.domain.fuzzy.etities.FuzzyProfile;
import aics.domain.fuzzy.models.FuzzyProfileData;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import lombok.experimental.Accessors;

import java.io.Serializable;

@Data
@Accessors(chain = true)
@RequiredArgsConstructor
public class FuzzyProfileDto implements Serializable {
    private Long fuzzyProfileId;
    private String name;
    private FuzzyProfileData fuzzyProfileData;
    private boolean showTopsisAnalysis;
    private boolean active;
    private boolean useFuzzyTopsis;

    public static FuzzyProfileDto fromFuzzyProfile(FuzzyProfile fuzzyProfile) {
        if (fuzzyProfile == null) {
            return null;
        }
        return new FuzzyProfileDto()
                .setFuzzyProfileId(fuzzyProfile.getFuzzyProfileId())
                .setName(fuzzyProfile.getName())
                .setFuzzyProfileData(fuzzyProfile.getFuzzyProfileData())
                .setShowTopsisAnalysis(fuzzyProfile.isShowTopsisAnalysis())
                .setActive(fuzzyProfile.isActive())
                .setUseFuzzyTopsis(fuzzyProfile.isUseFuzzyTopsis());
    }
}

File: semester2/5_Fuzzy-Software-Technologies/aics-fuzzy-software-technologies/ticket-frontend/src/modules/fuzzy/dtos/fuzzy-profile-dto.ts
import { FuzzyProfileData } from "../models/fuzzy-profile-data";

export class FuzzyProfileDto {
    fuzzyProfileId: number | null;
    name: string;
    fuzzyProfileData: FuzzyProfileData;
    showTopsisAnalysis: boolean;
    active: boolean;
    useFuzzyTopsis: boolean;

    constructor(obj: {
        fuzzyProfileId: number | null,
        name: string,
        fuzzyProfileData: FuzzyProfileData,
        showTopsisAnalysis: boolean,
        active: boolean,
        useFuzzyTopsis: boolean,
    }) {
        this.fuzzyProfileId = obj.fuzzyProfileId;
        this.name = obj.name;
        this.fuzzyProfileData = obj.fuzzyProfileData;
        this.showTopsisAnalysis = obj.showTopsisAnalysis;
        this.active = obj.active;
        this.useFuzzyTopsis = obj.useFuzzyTopsis;
    }

    static fromObj(obj: any): FuzzyProfileDto {
        if (!obj) {
            throw new Error('obj cannot be null');
        }
        return new FuzzyProfileDto({
            fuzzyProfileId: obj.fuzzyProfileId,
            name: obj.name,
            fuzzyProfileData: FuzzyProfileData.fromObj(obj.fuzzyProfileData),
            showTopsisAnalysis: obj.showTopsisAnalysis,
            active: obj.active,
            useFuzzyTopsis: obj.useFuzzyTopsis,
        });
    }

    deepClone(): FuzzyProfileDto {
        return new FuzzyProfileDto({
            fuzzyProfileId: this.fuzzyProfileId,
            name: this.name,
            fuzzyProfileData: this.fuzzyProfileData.deepClone(),
            showTopsisAnalysis: this.showTopsisAnalysis,
            active: this.active,
            useFuzzyTopsis: this.useFuzzyTopsis,
        })
    }
}